docs/data/material/customization/spacing/spacing.md

```diff
@@ -44,6 +44,44 @@ const theme = createTheme({
 theme.spacing(2); // = '8px'
 ```
 
+:::warning
+Note that when spacing is defined as an array, it only works with positive integers that will be used as array indexes.<br />
+It doesn't support all possible signatures of the `theme.spacing()` helper, for example `theme.spacing(0.5)`, `theme.spacing(-1)`, or `theme.spacing(1, 'auto')`.
+
+If you must use spacing array, consider using a function signature that can handle all possible signatures of the `theme.spacing()` helper:
+
+<details>
+<summary>Spacing function example</summary>
+
+```tsx
+const spacings = [0, 4, 8, 16, 32, 64];
+
+const theme = createTheme({
+  spacing: (factor: number | 'auto' = 0) => {
```

Comment:
Does `theme.spacing()` without any arguments match the current behavior of returning the first value in the spacing array (i.e., index 0)? So, in this case `theme.spacing()` would be `0`.

Reply:
- I think it's fine. Here, it's indicate that the spacing is a custom function so user has full control of the logic.

Reply:
Sorry for the late reply!

> Does theme.spacing() without any arguments here match with the current behavior? That is, it returns the first value in the spacing array (i.e., index 0)? Is this the current behviour?

Good point, `theme.spacing()` actually returns the **second** item in the array.
To me this is another argument that the array signature is unfortunate and confusing!